input/docs/fundamentals/criteria.md

```diff
@@ -6,20 +6,113 @@ You can control and influence the flow of the build script execution by providin
 ```csharp
 Task("A")
     .WithCriteria(DateTime.Now.Second % 2 == 0)
-    .Does(() =>
-{
-});
-
+    .Does(() => ... );
+    
 Task("B")
     .WithCriteria(() => DateTime.Now.Second % 2 == 0)
     .IsDependentOn("A")
-    .Does(() =>
-{
-});
+    .Does(() => ... );
 
 RunTarget("B");
 ```
 
 Task `A`'s criteria will be set when the task is created while Task `B`'s criteria will be evaluated when the task is being executed.
 
+In a real project, we would control the execution of the tasks as a pipeline according to different environmental states, the repository branches, etc. In this case we get two criterias from outside to detect if local build and we are working on master branch.
```

Comment:
You may want to call out the fact that you are specifically using AppVeyor, but you can do similar stuff with other Build Systems as well.